### PR TITLE
Add notification_url signature checking method

### PR DESCRIPTION
**Pull request by RTLcoil**

### CODE REVIEW

File: Shared/ApiShared.cs [hunk 1]
```diff
@@ -398,6 +398,51 @@ public string SignUriPart(string uriPart)
...
+        /// <returns>Boolean result of the validation</returns>
+        public bool VerifyNotificationSignature(string body, long timestamp, string signature, int validFor = 7200)
+        {
+            var currentTimestamp = Utils.UnixTimeNowSeconds();
+            var isSignatureExpired = timestamp <= currentTimestamp - validFor;
+            if (isSignatureExpired)
+                return false;
+
+            var payloadHash = Utils.ComputeHash($"{body}{timestamp}{Account.ApiSecret}");
+            var sign = new StringBuilder();
```
Comment (const-cloudinary) [inline Shared/ApiShared.cs]: I would be more reasonable to create another optional parameter like outputFormat for Utils.ComputeHash, or create another function ComputeHexHash, or create function bytesToHex or whatever and move hex conversion there.

File: Shared.Tests/ApiTest.cs [hunk 2]
```diff
@@ -1490,5 +1491,53 @@ public void TestSignParameters()
 
             StringAssert.AreNotEqualIgnoringCase(m_api.SignParameters(paramsSetOne), m_api.SignParameters(paramsSetTwo), "The signatures are equal.");
         }
+
+        [Test]
+        public void TestVerifyApiResponseSignature()
+        {
+            const string publicId = "resource id";
```
Comment (const-cloudinary) [inline Shared.Tests/ApiTest.cs]: We prefer to use the same test input values across SDKs to make sure we get the same output (if possible)

File: Shared.Tests/ApiTest.cs [hunk 3]
```diff
@@ -1490,5 +1491,53 @@ public void TestSignParameters()
...
+            responseParameters["public_id"] = "another id";
+            var anotherResourceSignature = m_api.SignParameters(responseParameters);
+
+            Assert.IsFalse(m_api.VerifyApiResponseSignature(publicId, version, anotherResourceSignature), "The response signature is invalid for the wrong resource");
+        }
+
+        [Test]
+        public void TestVerifyNotificationSignature()
+        {
+            const string body = "some response body";
```
Comment (const-cloudinary) [inline Shared.Tests/ApiTest.cs]: We prefer test values to be as similar as possible to the real world values.( in this case some JSON response)

File: Shared.Tests/UtilsTest.cs [hunk 4]
```diff
@@ -80,5 +80,17 @@ public void TestEncodeUrlSafeBase64String()
 
             Assert.AreEqual("YWQ_Lix4MDl-IUAhYQ==", Utils.EncodeUrlSafe("ad?.,x09~!@!a"));
         }
+
+        [Test]
+        public void TestComputeHexHash()
```
Comment (const-cloudinary) [inline Shared.Tests/UtilsTest.cs]: Please add reference input values and expected result. In case implementation is broken, these tests will still pass.

Comment (RTLcoil) [reply Shared.Tests/UtilsTest.cs]: fixed

**Pull request merged**